openaev-api/src/main/java/io/openaev/service/connector_instances/ConnectorInstanceLogService.java

```diff
@@ -0,0 +1,82 @@
+package io.openaev.service.connector_instances;
+
+import io.openaev.database.model.ConnectorInstance;
+import io.openaev.database.model.ConnectorInstanceLog;
+import io.openaev.database.repository.ConnectorInstanceLogRepository;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConnectorInstanceLogService {
+  @Getter private final Integer LOG_SIZE_LIMIT = 10000;
+  private final ConnectorInstanceLogRepository connectorInstanceLogRepository;
+
+  private void cleanupExcessLogs(String connectorInstanceId) {
+    long currentCount =
+        connectorInstanceLogRepository.countByConnectorInstanceId(connectorInstanceId);
+
+    if (currentCount > LOG_SIZE_LIMIT) {
+      int excessCount = (int) (currentCount - LOG_SIZE_LIMIT);
+      connectorInstanceLogRepository.deleteOldestLogByConnectorInstanceId(
+          connectorInstanceId, excessCount);
+      log.info("Deleted {} old logs for instance {}", excessCount, connectorInstanceId);
+    }
+  }
+
+  /**
+   * Transforms raw log lines into a single formatted log string.
+   *
+   * @param rawLogLines the set of raw log lines to transform
+   * @return the formatted log string with lines separated by newlines
+   */
+  public String transformRawLogsLineToLog(Set<String> rawLogLines) {
+    return rawLogLines.stream()
+        .map(line -> line.replaceAll("^,", ""))
+        .map(String::trim)
+        .filter(line -> !line.isEmpty())
+        .collect(Collectors.joining("\n"));
+  }
+
+  /**
+   * Creates a new log entry for a connector instance and maintains log limit.
+   *
+   * @param connectorInstance the connector instance to log for
+   * @param rawLog the log content to store
+   * @return the saved log entry
+   * @throws IllegalArgumentException if rawLog is empty
+   */
+  @Transactional
+  public ConnectorInstanceLog pushLogByConnectorInstance(
+      ConnectorInstance connectorInstance, String rawLog) throws IllegalArgumentException {
+    if (rawLog.isEmpty()) {
+      throw new IllegalArgumentException("Instance ID and log cannot be empty");
```

Comment:
Should we really fail with an exception here ? Maybe just doing nothing would be better ?